### PR TITLE
Add support for configuring terms and conditions in device commissioning

### PR DESCRIPTION
**Pull request by swan-amazon**
This commit introduces a new `enhanced_setup_flow` configuration property to the test harness backend, allowing device manufacturers to define the terms and conditions (T&C) that must be acknowledged and set during device commissioning. These values are critical for supporting the Terms and Conditions Certification Test Suite.

Key changes include:
- Added the `EnhancedSetupFlowConfig` class to `DutConfig`, which allows specifying the T&C version and user response required during commissioning.
- Updated `generate_command_arguments` in `utils.py` to append the necessary arguments for T&C validation and the device's commissioning method.
- Adjusted model validation to ensure the `enhanced_setup_flow` field is processed correctly within `DutConfig`.

These changes ensure the test harness can enforce manufacturer-specific T&C values during the commissioning process, aligning with the broader requirements of T&C validation.

### CODE REVIEW

File: app/user_prompt_support/uploaded_file_support.py [hunk 1]
```diff
@@ -30,8 +30,7 @@ class UploadFile(Protocol):
     filename: Optional[str]
 
     @property
-    def content_type(self) -> Optional[str]:
-        ...
+    def content_type(self) -> Optional[str]: ...
```
Comment (rquidute) [inline app/user_prompt_support/uploaded_file_support.py:33]:
```suggestion
    def content_type(self) -> Optional[str]:
        ...
```

File: test_collections/matter/sdk_tests/support/performance_tests/utils.py [hunk 2]
```diff
@@ -252,12 +252,12 @@ def generate_summary(
...
-    summary_dict["test_summary_record"][
-        "number_of_iterations_failed"
-    ] = compute_count_state(execution_status, False)
+    summary_dict["test_summary_record"]["number_of_iterations_passed"] = (
+        compute_count_state(execution_status, True)
+    )
+    summary_dict["test_summary_record"]["number_of_iterations_failed"] = (
+        compute_count_state(execution_status, False)
+    )
     summary_dict["test_summary_record"]["platform"] = "rpi"
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/performance_tests/utils.py:261]:
```suggestion
    summary_dict["test_summary_record"][
        "number_of_iterations_passed"
    ] = compute_count_state(execution_status, True)
    summary_dict["test_summary_record"][
        "number_of_iterations_failed"
    ] = compute_count_state(execution_status, False)
```
For some reason the `black`, changed this code formatting.

File: test_collections/matter/sdk_tests/support/python_testing/__init__.py [hunk 3]
```diff
@@ -34,6 +34,6 @@
         sdk_mandatory_python_test_collection()
     )
 
-    custom_python_collection: Optional[
-        TestCollectionDeclaration
-    ] = custom_python_test_collection()
+    custom_python_collection: Optional[TestCollectionDeclaration] = (
+        custom_python_test_collection()
+    )
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/python_testing/__init__.py:39]:
```suggestion
    custom_python_collection: Optional[
        TestCollectionDeclaration
    ] = custom_python_test_collection()
```
For some reason the `black`, changed this code formatting.

File: test_collections/matter/sdk_tests/support/python_testing/models/utils.py [hunk 4]
```diff
@@ -55,16 +55,28 @@ async def generate_command_arguments(
...
+
+    else:
         arguments.append(f"--commissioning-method {pairing_mode}")
 
-        if pairing_mode == DutPairingModeEnum.BLE_WIFI:
-            arguments.append(f"--wifi-ssid {config.network.wifi.ssid}")
-            arguments.append(f"--wifi-passphrase {config.network.wifi.password}")
+    if pairing_mode == DutPairingModeEnum.BLE_WIFI:
+        arguments.append(f"--wifi-ssid {config.network.wifi.ssid}")
+        arguments.append(f"--wifi-passphrase {config.network.wifi.password}")
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/python_testing/models/utils.py:75]:
```suggestion
        if pairing_mode == DutPairingModeEnum.BLE_WIFI:
            arguments.append(f"--wifi-ssid {config.network.wifi.ssid}")
            arguments.append(f"--wifi-passphrase {config.network.wifi.password}")
```
Could you please check this indentation, it seems it changes the code flow the way it was before.

File: test_collections/matter/sdk_tests/support/python_testing/models/utils.py [hunk 5]
```diff
@@ -55,16 +55,28 @@ async def generate_command_arguments(
...
 
-        if pairing_mode == DutPairingModeEnum.BLE_THREAD:
-            dataset_hex = await __thread_dataset_hex(config.network.thread)
-            arguments.append(f"--thread-dataset-hex {dataset_hex}")
+    if pairing_mode == DutPairingModeEnum.BLE_THREAD:
+        dataset_hex = await __thread_dataset_hex(config.network.thread)
+        arguments.append(f"--thread-dataset-hex {dataset_hex}")
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/python_testing/models/utils.py:79]:
```suggestion
        if pairing_mode == DutPairingModeEnum.BLE_THREAD:
            dataset_hex = await __thread_dataset_hex(config.network.thread)
            arguments.append(f"--thread-dataset-hex {dataset_hex}")
```
Could you please check this indentation, it seems it changes the code flow the way it was before.

File: test_collections/matter/test_environment_config.py [hunk 6]
```diff
@@ -99,6 +105,7 @@ def validate_model(self, dict_model: dict) -> None:
             # All DutConfig fields but chip_timeout are mandatory
```
Comment (rquidute) [inline test_collections/matter/test_environment_config.py:105]:
```suggestion
            # All DutConfig fields but chip_timeout and enhanced_setup_flow are mandatory
```

**Review by rquidute — CHANGES_REQUESTED** [review]
Please check some comments in the PR

**Comment by rquidute** [comment]
Hi @swan-amazon I changed the target from `main` to `v2.12`.
Please check my comments in the PR and also check the `Check spelling` and `Unit Tests` tasks which are failing. 
Thanks in advance.

File: app/user_prompt_support/uploaded_file_support.py (hunk 1 above)
Comment (swan-amazon) [reply app/user_prompt_support/uploaded_file_support.py:33]: @rquidute , I was getting additional CI failures due to black, so I locally ran the reformat... I think the repo is in a bad state. What should we do?

File: test_collections/matter/sdk_tests/support/performance_tests/utils.py (hunk 2 above)
Comment (swan-amazon) [reply test_collections/matter/sdk_tests/support/performance_tests/utils.py:261]: @rquidute , I was getting additional CI failures due to black, so I locally ran the reformat... I think the repo is in a bad state. What should we do?

File: test_collections/matter/sdk_tests/support/python_testing/__init__.py (hunk 3 above)
Comment (swan-amazon) [reply test_collections/matter/sdk_tests/support/python_testing/__init__.py:39]: @rquidute , I was getting additional CI failures due to black, so I locally ran the reformat... I think the repo is in a bad state. What should we do?

File: test_collections/matter/sdk_tests/support/python_testing/models/utils.py (hunk 4 above)
Comment (swan-amazon) [reply test_collections/matter/sdk_tests/support/python_testing/models/utils.py:75]: @rquidute , I was getting additional CI failures due to black, so I locally ran the reformat... I think the repo is in a bad state. What should we do?

File: test_collections/matter/sdk_tests/support/python_testing/models/utils.py (hunk 5 above)
Comment (swan-amazon) [reply test_collections/matter/sdk_tests/support/python_testing/models/utils.py:79]: @rquidute , I was getting additional CI failures due to black, so I locally ran the reformat... I think the repo is in a bad state. What should we do?

**Comment by swan-amazon** [comment]
> Hi @swan-amazon I changed the target from `main` to `v2.12`. Please check my comments in the PR and also check the `Unit Tests` tasks which is failing. The `Check spelling` task you don't need to worry about it. Thanks in advance.

Hey @rquidute , I'm not sure why those spell checks weren't failing previously. I didn't add the words that it's flagging on now... I was asking Raju but he didn't know. I asked if something was recently enabled and he doesn't think so. What should we do?

File: test_collections/matter/test_environment_config.py [hunk 7]
```diff
@@ -96,9 +102,10 @@ def validate_model(self, dict_model: dict) -> None:
                         f" {valid_properties}"
                     )
 
-            # All DutConfig fields but chip_timeout are mandatory
+            # All DutConfig fields but chip_timeout and enhanced_setup_flow are mandatory
```
Comment (rquidute) [inline test_collections/matter/test_environment_config.py:105]:
```suggestion
            # All DutConfig fields but chip_timeout and enhanced_setup_flow
            # are mandatories
```

Comment (rquidute) [reply test_collections/matter/test_environment_config.py:105]: Line was too long >88

File: test_collections/matter/sdk_tests/support/performance_tests/utils.py [hunk 8]
```diff
@@ -258,7 +258,6 @@ def generate_summary(
     summary_dict["test_summary_record"][
         "number_of_iterations_failed"
     ] = compute_count_state(execution_status, False)
-    summary_dict["test_summary_record"]["platform"] = "rpi"
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/performance_tests/utils.py:261]: It seems this line was removed unintentionally

**Review by rquidute — CHANGES_REQUESTED** [review]
By now, just minor items and also the Unit Tests must be fixed.

**Comment by rquidute** [comment]
> > Hi @swan-amazon I changed the target from `main` to `v2.12`. Please check my comments in the PR and also check the `Unit Tests` tasks which is failing. The `Check spelling` task you don't need to worry about it. Thanks in advance.
> 
> Hey @rquidute , I'm not sure why those spell checks weren't failing previously. I didn't add the words that it's flagging on now... I was asking Raju but he didn't know. I asked if something was recently enabled and he doesn't think so. What should we do?

Hi @swan-amazon don't worry about spelling issue in this PR. There is still just one spelling issue, but this is a temp code. This is something we need to fix.

File: test_collections/matter/test_environment_config.py (hunk 7 above)
Comment (swan-amazon) [reply test_collections/matter/test_environment_config.py]: Fixed

File: test_collections/matter/sdk_tests/support/performance_tests/utils.py (hunk 8 above)
Comment (swan-amazon) [reply test_collections/matter/sdk_tests/support/performance_tests/utils.py]: Reverted changes to this file.

**Review by rquidute — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

**Comment by rquidute** [comment]
Hi @swan-amazon PR is approved. 
Could you please update the User Guide according to this new added feature?

**Pull request merged**